src/components/course/EnrollButton.jsx

```diff
@@ -108,20 +111,48 @@ export default function EnrollButton() {
     </EnrollButtonWrapper>
   );
 
+  const enrollBtnDisabled = (
+    <EnrollButtonCta
+      as="div"
+      className="btn btn-light btn-block disabled"
+    />
+  );
+
   const enrollLinkClass = 'btn-block';
 
   if (!userEnrollment && isEnrollable) {
     // enroll with a subscription license
     if (enrollmentUrl && subscriptionLicense) {
+      if (userSubsidy) {
+        return (
+          <EnrollButtonCta
+            as="a"
+            className={classNames('btn btn-primary btn-brand-primary', enrollLinkClass)}
+            href={enrollmentUrl}
+          />
+        );
+      }
+      // no user subsidy means we need to warn user with a dialog
       return (
-        <EnrollButtonCta
-          as="a"
-          className={classNames('btn btn-primary btn-brand-primary', enrollLinkClass)}
-          href={enrollmentUrl}
-        />
+        <>
+          <EnrollButtonCta
+            className={enrollLinkClass}
+            onClick={() => setIsModalOpen(true)}
+          />
+          <EnrollModal
+            isModalOpen={isModalOpen}
+            setIsModalOpen={setIsModalOpen}
+            offersCount={offersCount}
+            courseHasOffer={!!findOfferForCourse(offers, catalogList)}
+            enrollmentUrl={enrollmentUrl}
+          />
+        </>
       );
     }
     // enroll with an offer (code)
+    if (!features.ENROLL_WITH_CODES) { // can't do offer based enrollment yet without switch
+      return enrollBtnDisabled;
+    }
     if (enrollmentUrl) {
```

Comment:
Just to confirm here, if we reach the conditional block on this line 156, we know the user has no user subsidy (license and/or code) so the enroll button will open the modal which brings user to the basket to enter payment. Is that correct? (Still catching up a bit on the codes-related changes to the enrollment flow haha)

Reply:
correct.

however, now this makes me think, is this a valid state?

    - no subscriptionLicense, but still valid userSubsidy?

if thatt is the case currently looks like we will still pop a modal and the enrollmenturl will be basket one. I can locally test but reading through it seems that way.

Is that okay? the tricky part about the code here is we don't exactly know whatt enrollmenturl wil lbe here without referring to the other logic.. I plan to bring them together in the refactor.

Reply:
discussed with @adamstankiewicz on slack, it appears that the only subsidy type addressed right now is the license type as per https://github.com/edx/frontend-app-learner-portal-enterprise/blob/f11b0aa4d0d14277c18d9faf4c70b6b6536f1a45/src/components/course/data/service.js#L113 and this logic here is still okay since an invalid userSubsidy seems correlated right now, to an invalid subscription

so for example one could get to this line without a valid subscription but still may have a valid code subsidy, and that is okay since we already handle that case int he url generation logic